iis/mymodule.cpp

```diff
@@ -1140,7 +1125,9 @@ CMyHttpModule::OnBeginRequest(
 #endif
 	c->remote_host = NULL;
 
+    LeaveCriticalSection(&m_csLock);
 	int status = modsecProcessRequest(r);
```

Comment:
Could we add in the description, what tests have we done ?

Reply:
Tested by running running 80 difficult requests in parallel overnight, and ensuring no requests that there were no mixup between 200 and 403.